### PR TITLE
[PS] Browser nav bar facelift

### PR DESCRIPTION
**Pull request by jpcloureiro**
**Description**

Update browser nav bar

**Issue**

Progresses #4762 

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/BrowserUrlBar/BrowserUrlBar.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  const isHttps = url && url.toLowerCase().substr(0, 6) === 'https:';
+
+  const secureConnectionIcon = isHttps
+    ? IconName.LockFilled
+    : IconName.LockSlashFilled;
+
+  const mainUrl = getDappMainUrl();
+
+  const { styles, theme } = useStyles(stylesheet, {});
```
Comment (blackdevelopa) [inline app/components/UI/BrowserUrlBar/BrowserUrlBar.tsx:38]: I didn't know we can do this. Looks cool

Comment (jpcloureiro) [reply app/components/UI/BrowserUrlBar/BrowserUrlBar.tsx:38]: useStyles is a new hook we created for the component-library components. It is useful when you need to build a stylesheet with conditionals, use theme colors, all in a single `styles.ts` file

File: app/components/UI/AccountRightButton/index.tsx [hunk 2]
```diff
@@ -47,7 +47,10 @@ const AccountRightButton = ({
     (state: any) => state.engine.backgroundState.NetworkController.provider,
   );
   const dispatch = useDispatch();
+  // TODO: change onPressButton to a const, using a ternary
+
```
Comment (Cal-L) [inline app/components/UI/AccountRightButton/index.tsx:51]: Remove comments

File: app/components/Views/Browser/index.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ const Browser = (props) => {
   useEffect(
     () => {
       navigation.setOptions(
-        getBrowserViewNavbarOptions(navigation, route, drawerRef, colors),
+        getBrowserViewNavbarOptions(route, colors),
```
Comment (Cal-L) [inline app/components/Views/Browser/index.js:50]: remove drawer ref

File: app/components/UI/BrowserUrlBar/BrowserUrlBar.types.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export interface BrowserUrlBarProps {
+  url: string;
+  route: any;
```
Comment (Cal-L) [inline app/components/UI/BrowserUrlBar/BrowserUrlBar.types.ts:3]: Let's provide it with the known type

File: app/components/UI/BrowserUrlBar/BrowserUrlBar.styles.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { StyleSheet } from 'react-native';
+
+import { Theme } from '../../../util/theme/models';
+
+const styleSheet = (params: { theme: Theme }) =>
+  StyleSheet.create({
+    main: {
+      flexDirection: 'row',
+      marginLeft: 23,
```
Comment (Cal-L) [inline app/components/UI/BrowserUrlBar/BrowserUrlBar.styles.ts:9]: update to marginHorizontal: 16

File: app/components/UI/BrowserUrlBar/BrowserUrlBar.styles.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      marginRight: 23,
+    },
+    text: {
+      flex: 1,
+      marginLeft: 6,
+      color: params.theme.colors.icon.alternative,
```
Comment (Cal-L) [inline app/components/UI/BrowserUrlBar/BrowserUrlBar.styles.ts:15]: colors.text.alternative

File: app/components/UI/BrowserUrlBar/BrowserUrlBar.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+const BrowserUrlBar = ({ url, route, onPress }: BrowserUrlBarProps) => {
+  const getDappMainUrl = () => {
+    if (!url) return;
+
+    const urlObj = new URL(url);
+    const ensUrl = route.params?.currentEnsName ?? '';
+
+    if (
+      isGatewayUrl(urlObj) &&
```
Comment (Cal-L) [inline app/components/UI/BrowserUrlBar/BrowserUrlBar.tsx:21]: In future iterations, we can clean up this code

**Review by Cal-L — APPROVED** [review]
After comments, LGTM

**Pull request merged**